qa/tasks/cephfs/kernel_mount.py

```diff
@@ -51,6 +51,9 @@ def mount(self, mount_path=None, mount_fs_name=None):
         if mount_fs_name is not None:
             opts += ",mds_namespace={0}".format(mount_fs_name)
 
+        for mount_opt in mount_options :
+            opts += ",{0}".format(mount_opt)
+
```

Comment:
Now that this method fails when the options are unsupported, we can have a `mnt.[0-9]` dir left behind which causes subsequent tests to fail. We need to catch the exception when running the mnt and `rmdir` the mount point, then raise the exception again for the caller to deal with.

````
2019-11-13T19:01:34.320 INFO:teuthology.orchestra.run.smithi001:Running:
2019-11-13T19:01:34.320 INFO:teuthology.orchestra.run.smithi001:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage /bin/mount -t ceph :/ /home/ubuntu/cephtest/mnt.0 -v -o name=0,norequire_active_mds,conf=/etc/ceph/ceph.conf,recover_session=clean
2019-11-13T19:01:34.408 DEBUG:teuthology.orchestra.run:got remote process result: 32
2019-11-13T19:01:34.440 INFO:teuthology.orchestra.run.smithi001.stderr:mount error 22 = Invalid argument
2019-11-13T19:01:34.441 INFO:teuthology.orchestra.run.smithi001.stdout:parsing options: rw,name=0,norequire_active_mds,conf=/etc/ceph/ceph.conf,recover_session=clean
2019-11-13T19:01:34.441 INFO:teuthology.orchestra.run.smithi001.stdout:mount.ceph: unrecognized mount option "norequire_active_mds", passing to kernel.
2019-11-13T19:01:34.441 INFO:teuthology.orchestra.run.smithi001.stdout:mount.ceph: unrecognized mount option "recover_session", passing to kernel.
2019-11-13T19:01:34.441 INFO:teuthology.orchestra.run.smithi001.stdout:parsing options: rw,name=0,norequire_active_mds,conf=/etc/ceph/ceph.conf,recover_session=clean
2019-11-13T19:01:34.441 INFO:teuthology.orchestra.run.smithi001.stdout:mount.ceph: unrecognized mount option "norequire_active_mds", passing to kernel.
2019-11-13T19:01:34.441 INFO:teuthology.orchestra.run.smithi001.stdout:mount.ceph: unrecognized mount option "recover_session", passing to kernel.
2019-11-13T19:01:34.442 INFO:tasks.cephfs_test_runner:test_reconnect_after_blacklisted (tasks.cephfs.test_client_recovery.TestClientRecovery) ... skipped 'Not implemented in current kernel'
...
2019-11-13T19:01:53.994 INFO:teuthology.orchestra.run.smithi001.stderr:mkdir: cannot create directory ‘/home/ubuntu/cephtest/mnt.0’: File exists
2019-11-13T19:01:53.995 INFO:tasks.cephfs_test_runner:test_reconnect_eviction (tasks.cephfs.test_client_recovery.TestClientRecovery) ... ERROR
2019-11-13T19:01:53.995 INFO:tasks.cephfs_test_runner:
2019-11-13T19:01:53.995 INFO:tasks.cephfs_test_runner:======================================================================
2019-11-13T19:01:53.995 INFO:tasks.cephfs_test_runner:ERROR: test_reconnect_eviction (tasks.cephfs.test_client_recovery.TestClientRecovery)
2019-11-13T19:01:53.995 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191112.231923/qa/tasks/cephfs/cephfs_test_case.py", line 148, in setUp
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:    self.mounts[i].mount()
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191112.231923/qa/tasks/cephfs/kernel_mount.py", line 42, in mount
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:    timeout=(5*60),
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 205, in run
2019-11-13T19:01:53.996 INFO:tasks.cephfs_test_runner:    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 437, in run
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:    r.wait()
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 162, in wait
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:    self._raise_for_status()
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 184, in _raise_for_status
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:    node=self.hostname, label=self.label
2019-11-13T19:01:53.997 INFO:tasks.cephfs_test_runner:CommandFailedError: Command failed on smithi001 with status 1: 'mkdir -- /home/ubuntu/cephtest/mnt.0'
````


/ceph/teuthology-archive/pdonnell-2019-11-13_04:31:08-kcephfs-wip-pdonnell-testing-20191112.231923-distro-basic-smithi/4502503/teuthology.log